MBBSEmu/CPU/CPUCore.cs

```diff
@@ -1665,17 +1669,17 @@ private byte Op_Rcl_8()
                 var result = destination;
                 for (var i = 0; i < source; i++)
                 {
-                    //Determine the CF Value after rotation+carry
+                    // Determine the CF Value after rotation+carry
                     var newCFValue = result.IsBitSet(7);
 
-                    //Perform Rotation
+                    // Perform Rotation
                     result = (byte)(result << 1);
 
-                    //If CF was set, rotate that value in
+                    // If CF was set, rotate that value in to the LSB (0th bit)
                     if (Registers.CarryFlag)
-                        result.SetFlag(1);
+                        result |= 1;
```

Comment:
isn't this equivalent?

Reply:
No, technically `.SetFlag` is an extension method we have on `byte` types, which since they're static, can't modify the underlying instance value. This was one of the bugs with the underlying logic here, because if using the helper method it should be `result = result.SetFlag(1)`, which I felt was overhead for just a ` |= 1` operation.

Reply:
ah, subtle